### PR TITLE
Show Public URL for documents in backend

### LINKED ISSUE #10832: [Improvement]: Show Public URL for documents in backend
### Improvement description

Similar as we already have it for assets.
image

Would be helpful in some situations to quickly retrieve the URL of a document (without having to open it in the frontend).

Should also consider pretty URLs & sites.

### PR DESCRIPTION
**Pull request by jelena-hrga**
Resolves #10832
Added public URL and pretty URL on document info popup

### CODE REVIEW

**Comment by brusch** [comment]
Thanks for your PR! 
I think we should have just one field für the public URL. If there's a pretty URL, this should be used, if not, the path of the document. I think we'd need to fix this here: https://github.com/pimcore/pimcore/blob/36a44b071f98d6e632312e6e797006ef4af20f94/models/Document/PageSnippet.php#L600

Can you please change your PR accordingly? 
Thanks a lot!

**Comment by jelena-hrga** [comment]
@brusch Thank you for the feedback, I changed my PR accordingly.

File: models/Document/PageSnippet.php [hunk 1]
```diff
@@ -597,7 +597,8 @@ public function getUrl($hostname = null, $scheme = null)
             }
         }
 
-        $url = $scheme . $hostname . $this->getFullPath();
+        $url = $scheme . $hostname;
+        $url .= $this->getPrettyUrl() ?: $this->getFullPath();
```
Comment (brusch) [inline models/Document/PageSnippet.php]:
```suggestion
        if($this instanceof Page && $this->getPrettyUrl()) {
            $url .= $this->getPrettyUrl();
        } else {
            $url .= $this->getFullPath();
        }
```

**Review by brusch — CHANGES_REQUESTED** [review]

File: models/Document/PageSnippet.php (hunk 1 above)
Comment (brusch) [reply models/Document/PageSnippet.php]: We need to check if `$this` is a `Page` because `Snippet` doesn't have the method `getPrettyUrl()`.

**Comment by brusch** [comment]
Thank you very much!

**Pull request merged**